src/atomate2/abinit/schemas/core.py

```diff
@@ -0,0 +1,212 @@
+)
+from atomate2.common.schemas.structure import StructureMetadata
+
+_T = TypeVar("_T", bound="AbinitTaskDocument")
+
+logger = logging.getLogger(__name__)
+
+
+class Status(ValueEnum):
+    """Abinit calculation state."""
+
+    # TODO: merge this somewhere with vasp => common calculation schema ?
+
+    SUCCESS = "successful"
+    FAILED = "failed"
+
+
+class JobMetadata(BaseModel):
+    """Definition of job metadata fields."""
+
+    dir_name: str = Field(None, description="The directory of this job.")
+    calc_type: str = Field(None, description="The type of calculation of this job.")
+
+
+class AbinitJobSummary(JobMetadata):
+    """Definition of summary information about an Abinit Job."""
+
+    # restart_info: RestartInfo = Field(
+    #     None, description="Restart information for the next job."
+    # )
+    # history: JobHistory = Field(None, description="Job history.")
+    abinit_input_set: AbinitInputSet = Field(
+        None, description="AbinitInputSet object used to perform calculation."
+    )
+    structure: Structure = Field(
+        None, description="Final structure of the calculation."
+    )
+    energy: float = Field(None, description="Final energy of the calculation.")
+
+
+class AbinitTaskDocument(StructureMetadata):
+    """Definition of task document about an Abinit Job."""
+
+    state: Status = Field(None, description="State of this job.")
+    run_number: int = Field(None, description="Run number of this job.")
+    dir_name: str = Field(None, description="The directory of this job.")
+    event_report: events.EventReport = Field(
+        None, description="Event report of this abinit job."
+    )
+    task_label: str = Field(None, description="The label for this job/task.")
+    abinit_input: AbinitInput = Field(
+        None, description="AbinitInput used to perform calculation."
+    )
+
+    @classmethod
+    def from_directory(
+        cls: Type[_T],
+        dir_name: Union[Path, str],
+        source: str = "log",
+        critical_events=None,
+        run_number=1,
+        structure_fixed=True,
+    ):
+        """Build AbinitTaskDocument from directory."""
+        # Files required for the job analysis.
+        # TODO: See if we can put the AbinitInputFile object here from
+        #  abipy.abivars.AbinitInputFile (currently not MSONable)
+        # input_file = File(os.path.join(dir_name, INPUT_FILE_NAME))
+        output_file = File(os.path.join(dir_name, OUTPUT_FILE_NAME))
+        log_file = File(os.path.join(dir_name, LOG_FILE_NAME))
+        mpiabort_file = File(os.path.join(dir_name, MPIABORTFILE))
+        ofile = {"output": output_file, "log": log_file}[source]
+
+        report = None
+        state = Status.FAILED
+
+        try:
+            report = cls.get_event_report(ofile=ofile, mpiabort_file=mpiabort_file)
+            critical_events_report = report.filter_types(critical_events)
+            if not critical_events_report:
+                state = Status.SUCCESS
```

Comment:
About 1., why do you say there is part of the restarting logic there ? You mean with the status ? Because I would say the logic of the restart is still in the get_response_args method of the maker.
About 2. indeed, in Vasp, when it is not converged, it is failed, because there is no restart possibility. We can consider to have SUCCESS, FAILED and UNCONVERGED ?

Reply:
For 1. let's keep it like that.
For 2. I have added UNCONVERGED. I still have to detect whether it is failed or unconverged. In the TaskDocument, I would indeed keep it as failed because it is the status of the calculation, the subsequent job will be the one that is SUCCEEDED.

Reply:
What do you think ?